framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -3401,6 +3451,12 @@ void VulkanReplayConsumerBase::OverrideFreeMemory(PFN_vkFreeMemory  func,
         memory         = memory_info->handle;
         allocator_data = memory_info->allocator_data;
 
+        auto findIt = external_memory_.find(memory);
+        if (findIt != external_memory_.end())
+        {
+            util::platform::FreeRawMemory(findIt->second.first, findIt->second.second);
+        }
+
```

Comment:
Does it need to erase memory from external_memory_?

Reply:
Well, didn't found any clear information about this in vulkan spec's

At least gfxreconstruct already do so in the vulkan_capture_manager.cpp VulkanCaptureManager::PostProcess_vkFreeMemory function

Also KhronosGroup VK-GL-CTS test erases the memory as well, so I think it's needed action

https://github.com/KhronosGroup/VK-GL-CTS/blob/60be5c346e5a63d45ca55a938e1025c0c5d32432/external/vulkancts/modules/vulkan/memory/vktMemoryExternalMemoryHostTests.cpp

```
ExternalMemoryHostBaseTestInstance::~ExternalMemoryHostBaseTestInstance (void)
{
	deAlignedFree(m_hostMemoryAlloc);
}
```

Reply:
Done

Reply:
I can squash all into a single commit if you'd like

Reply:
Squash would be better. Thank you very much.

Reply:
Done